### PR TITLE
Capability injection

### PR DESCRIPTION
**Pull request by marcvberg**
#### Description of changes

This change enables AWS resource provisioning capability insertion for the vNext GraphQL Transformer

#### Description of how you validated changes
Created a new project and generated the different types of resources in the vNext transformer to validate that moving them to a new class didn't break them

#### Checklist

- [x] PR description included
- [x] `yarn test` passes

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-graphql-transformer-interfaces/src/transform-host-provider.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
+import {Duration, Stack} from "@aws-cdk/core";
```
[the remark this answers is not in the document]
Comment (marcvberg) [reply packages/amplify-graphql-transformer-interfaces/src/transform-host-provider.ts:1]: Got it, I'll update it

File: packages/amplify-graphql-transformer-core/src/transformation/transform.ts [hunk 2]
```diff
@@ -36,6 +36,7 @@ import {
   sortTransformerPlugins,
 } from './utils';
 import { validateModelSchema } from './validation';
+import {TransformHostProvider} from "@aws-amplify/graphql-transformer-interfaces/lib/transform-host-provider";
```
[the remark this answers is not in the document]
Comment (marcvberg) [reply packages/amplify-graphql-transformer-core/src/transformation/transform.ts:39]: Sure, I'll get that fixed

**Comment by marcvberg** [comment]
@cjihrig Should be good now, resolved linting errors

File: packages/amplify-graphql-transformer-core/src/graphql-api.ts [hunk 3]
```diff
@@ -1,36 +1,22 @@
...
-import { CfnFunction, Code, Function, IFunction, ILayerVersion, Runtime } from '@aws-cdk/aws-lambda';
-import { CfnResource, Construct, Duration, Stack, Token } from '@aws-cdk/core';
-import { toCamelCase } from 'graphql-transformer-common';
-import { AppSyncFunctionConfiguration } from './appsync-function';
-import { ElasticsearchDataSource } from './cdk-compat/elasticsearch-datasource';
+import { Grant, IGrantable, ManagedPolicy, Role, ServicePrincipal } from '@aws-cdk/aws-iam';
+import { CfnResource, Construct, Duration, Stack } from '@aws-cdk/core';
 import { TransformerSchema } from './cdk-compat/schema-asset';
-import { InlineTemplate, S3MappingFunctionCode } from './cdk-compat/template-asset';
+import {DefaultTransformHost} from './transform-host';
```
Comment (cjihrig) [inline packages/amplify-graphql-transformer-core/src/graphql-api.ts:19]:
```suggestion
import { DefaultTransformHost } from './transform-host';
```

File: packages/amplify-graphql-transformer-interfaces/src/transform-host-provider.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
+import {Duration, Stack} from '@aws-cdk/core';
```
Comment (cjihrig) [inline packages/amplify-graphql-transformer-interfaces/src/transform-host-provider.ts:1]:
```suggestion
import { Duration, Stack } from '@aws-cdk/core';
```

There are a few more of these below.

File: packages/amplify-graphql-transformer-core/src/transform-host.ts [hunk 5]
```diff
@@ -0,0 +1,301 @@
+import {TransformHostProvider} from '@aws-amplify/graphql-transformer-interfaces';
```
Comment (cjihrig) [inline packages/amplify-graphql-transformer-core/src/transform-host.ts:1]:
```suggestion
import { TransformHostProvider } from '@aws-amplify/graphql-transformer-interfaces';
```

There are a few more of these below.

**Review by cjihrig — APPROVED** [review]
Some very minor formatting comments, but LGTM

**Pull request merged**